### DISCUSSION TITLE
Support for R, Shiny

### DISCUSSION THREAD

**Discussion by Ed2uiz** [uwdata/mosaic#354] [General]
Hello,

Is it possible to use `mosaic` with R or Shiny? If not, are there any plans to make this possible? I know that `vega` and `altair` have made their way into R packages (vegalite, vegabrite, etc...) through the `htmlwidgets` framework, though I'm not sure how relevant they are to making `mosaic` compatible with R. Regardless, it would be great to be able to use mosaic visualizations through R at some point and look forward to reading any feedback or pointers on how to make this happen.

Thanks,
Ed

**Comment by jheer** [comment]
Thanks for asking! Unfortunately I'm not familiar enough with the R ecosystem to say. In theory it should be possible if Shiny (or something similar) supports message passing (e.g., over a web socket connection) from a browser-based component to a custom R handler (which would proxy access to a DuckDB instance running in R). This is the basic architecture that the Mosaic Jupyter widget for Python follows. Maybe others more knowledgeable about R can chime in...

**Comment by Ed2uiz** [comment]
Hello @jheer, I was able to work on this project alongside a talented student @TiRizvanov via the [rMosaic](https://github.com/TiRizvanov/rMosaic) package, which is now on [CRAN](https://cran.r-project.org/web/packages/rMosaic/index.html)! It's missing a proper R-native spec generating API, but it supports `htmlwidgets` and `shiny` via multiple spec formats. Also copying @rpruim and @kcarnold from #929 in case there's still interest, feedback + contributions welcome. Thanks!

**Reply by jheer** [reply]
> Hello @jheer, I was able to work on this project alongside a talented student @TiRizvanov via the [rMosaic](https://github.com/TiRizvanov/rMosaic) package, which is now on [CRAN](https://cran.r-project.org/web/packages/rMosaic/index.html)! It's missing a proper R-native spec generating API, but it supports `htmlwidgets` and `shiny` via multiple spec formats. Also copying @rpruim and @kcarnold from #929 in case there's still interest, feedback + contributions welcome. Thanks!

That's fantastic. Very glad to see this!
